src/driver/socket.rs

```diff
@@ -40,12 +40,12 @@ impl Socket {
     }
 
     pub(crate) async fn write<T: IoBuf>(&self, buf: T) -> crate::BufResult<usize, T> {
-        let op = Op::write_at(&self.fd, buf, 0).unwrap();
+        let op = Op::write_at(&self.fd, buf, 0);
```

Comment:
I believe the `async` transformation works so that this is not run until the future returned by `write` is polled for the first time. If `Op` creation only happens in such places, we have nothing to worry about?

Reply:
I don't follow, sorry

Reply:
Line 43 is not executed synchronously, that happens on the first poll of the future returned by `write`. So this method does not actually do any work until polled.
Is there any instance where `submit_with` is called inside a synchronous fn?